stacker/hooks/keypair.py

```diff
@@ -24,18 +37,28 @@ def ensure_keypair_exists(provider, context, **kwargs):
         provider (:class:`stacker.providers.base.BaseProvider`): provider
             instance
         context (:class:`stacker.context.Context`): context instance
+        keypair (:class:` str'): name of the key pair to create
+        ssm_parameter_name (:class:`str`): optional path to an SSM store
+            parameter to receive the generated private key, instead of
+            importing it or storing it locally.
+        ssm_key_id (:class:`str`): optional ID of a KMS key to encrypt the SSM
+            parameter with. If ommitted, the default key will be used.
 
     Returns: boolean for whether or not the hook succeeded.
-
     """
+
+    keypair_name = kwargs["keypair"]
+    ssm_parameter_name = kwargs.get("ssm_parameter_name")
```

Comment:
I'd make both of these optional arguments (ie: default to `None`) in the function signature.